app/src/main/java/cm/aptoide/pt/home/HomePresenter.java

```diff
@@ -86,6 +98,27 @@ public HomePresenter(HomeView view, Home home, Scheduler viewScheduler, CrashRep
     handleClickOnPrivacyPolicy();
 
     handleEditorialCardClick();
+
+    handleAppNextAdClick();
+  }
+
+  private void handleAppNextAdClick() {
+    view.getLifecycleEvent()
+        .filter(event -> event.equals(View.LifecycleEvent.CREATE))
+        .flatMap(__ -> adsManager.appNextAdClick()
+            .observeOn(Schedulers.io())
+            .flatMap(result -> {
+              AppNextNativeAd ad = result.getAd();
+              homeAnalytics.sendAdClickEvent(ad.getStars(), ad.getPackageName(), 0,
+                  "ads-highlighted", HomeEvent.Type.AD, ApplicationAd.Network.APPNEXT);
+              return highlightedAdExperiment.recordAdClick();
+            }))
+        .retry()
```

Comment:
this retry is on the lifecycle event, if it fails it will not wait for an appnext ad click unless a lifecycle event occurs before.